### PR TITLE
pkg/kubeapiserver/options: Improving test coverage

### PR DESCRIPTION
**Pull request by tmilox**
Signed-off-by: TommyStarK <thomasmilox@gmail.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

If applied this commit will increase the unit tests code coverage of `pkg/kubeapiserver/options`

- Initial unit tests code coverage

Screenshot 2022-12-01 at 13 08 39

- After changes

Screenshot 2022-12-01 at 13 37 37

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by TommyStarK** [comment]
Hello @cheftako, @liggitt 

I know guys you are super busy and this PR is far from being a priority. As a very new aspiring contributor I am still trying to become familiar with the code and to start helping on small pieces. Hope you don't mind. 

Any feedback will be greatly appreciated 😅.

Best regards,

**Review by jpbetz — APPROVED** [review]
/lgtm

Test coverage increase looks great.

I did a pass on the tests and they made sense to me.

**Comment by TommyStarK** [comment]
Hello @jpbetz 

Thank you for the review :) I updated my PR to correct the two misspelling that made the kubernetes-verify job to fail.

Screenshot 2022-12-02 at 10 59 16


Best,

**Comment by TommyStarK** [comment]
@jpbetz , @aimuz 

All pipelines have passed, don't hesitate if there is anything else I can do.

Best regards,

**Comment by TommyStarK** [comment]
Hello  @deads2k , @liggitt

@jpbetz Thank you very much for your time!

I know guys you are super busy and this is PR is far from a being a priority. Just wanted to give it a last try to get a review/approval.

Best regards,

**Comment by TommyStarK** [comment]
Hello @jpbetz 

I updated my PR to include a suggestion made by @liggitt on another PR. Added a snippet of the expected error string related to the aspect being made invalid to ensure that the config is not getting rejected for an unrelated reason.

Remaining at your disposal.

Best regards,

File: pkg/kubeapiserver/options/authentication_test.go [hunk 1]
```diff
@@ -161,7 +221,7 @@ func TestToAuthenticationConfig(t *testing.T) {
 			ClientCA: "testdata/root.pem",
 		},
 		WebHook: &WebHookAuthenticationOptions{
-			CacheTTL:   180000000000,
+			CacheTTL:   1 * time.Second,
```
Comment (liggitt) [inline pkg/kubeapiserver/options/authentication_test.go:224]: curious why the cache ttls were changed

**Review by liggitt — COMMENTED** [review]
one question about the TTL changes, lgtm otherwise

File: pkg/kubeapiserver/options/authentication_test.go (hunk 1 above)
Comment (liggitt) [reply pkg/kubeapiserver/options/authentication_test.go:224]: yeah, if we're not actually checking the warnings in the test, let's leave these lines unmodified for now

Comment (TommyStarK) [reply pkg/kubeapiserver/options/authentication_test.go]: Done :) If there is anything else please tell me.
Thank you @liggitt

**Pull request merged**